front/learn-next-js/src/app/components/ScheduleTable.tsx

```diff
@@ -0,0 +1,46 @@
+import { promises as fs } from 'fs';
+
+
+async function CreateTable(schedule: []) {
+    const scheduleEnries = Object.entries(schedule)
+    const firstDayInSchedule = scheduleEnries[0]
+    const scheduleDates = scheduleEnries.map((day) => day[0])
+  
+    const tHeaders = scheduleDates.map((header) => <th className="p-1 text-black">{header}</th>)
+    // the firstDayInDictionary is a list containing a stringify date and a dictionary, the dictionary containe a "shift type" as a key (string) and a string representing the employee and the shift's period of time
```

Comment:
this comment is not bad, but use type script to define the types in addition to explaining what the objects contain